src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,286 @@
+                throwWrongToDoFormatException();
+            }
+            break;
+        case DEADLINE:
+            if (indexOfSpace == -1) {
+                throwEmptyDescriptionException();
+            } else if (messageArr.length < 2) {
+                throwWrongDeadlineFormatException();
+            } else if (messageArr[1].length() < 4) {
+                throwEmptyDateException();
+            }
+            break;
+        case EVENT:
+            if (indexOfSpace == -1) {
+                throwEmptyDescriptionException();
+            } else if (messageArr.length < 4) {
+                throwWrongEventFormatException();
+            } else if (messageArr[1].length() < 4) {
+                throwEmptyDateException();
+            }
+            break;
+        default:
+            throw new DukeException("Unexpected value: " + type);
+        }
+    }
+    private LocalDate parseDateFromString(String dateString, TaskTypes type) throws DukeException {
+        try {
+            return LocalDate.parse(dateString, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+        } catch (DateTimeException e) {
+            if (type == TaskTypes.DEADLINE) {
+                throwWrongDeadlineFormatException();
+            } else {
+                throwWrongEventFormatException();
+            }
+        }
+        return null; //should not reach here
+    }
+    private LocalTime parseTimeFromString(String timeString, TaskTypes type) throws DukeException {
+        try {
+            return LocalTime.parse(timeString, DateTimeFormatter.ofPattern("HH:mm"));
+        } catch (DateTimeException e) {
+            if (type == TaskTypes.DEADLINE) {
+                throwWrongDeadlineFormatException();
+            } else {
+                throwWrongEventFormatException();
+            }
+        }
+        return null; //should not reach here
+    }
+
+    private void throwWrongToDoFormatException() throws DukeException {
+        throw new DukeException("Pardon me, but the description of a todo cannot be empty.");
+    }
+    private void throwWrongDeadlineFormatException() throws DukeException {
+        throw new DukeException("Pardon me, but the Deadline format is incorrect."
+            + " The format should be:\n\t[Task] [Description] /by yyyy-mm-dd/HH:mm (leave \"/HH:mm\""
+            + " empty if no time in current task)");
+    }
+
+    private void throwWrongEventFormatException() throws DukeException {
+        throw new DukeException("Pardon me, but the Event format is incorrect."
+            + " The format should be:\n\t[Task] [Description] /at yyyy-mm-dd/HH:mm/HH:mm");
+    }
+    private void throwEmptyDescriptionException() throws DukeException {
+        throw new DukeException("Pardon me, but the description of a deadline/event cannot be empty.");
+    }
+
+    private void throwEmptyDateException() throws DukeException {
+        throw new DukeException("Pardon me, but the date/time cannot be empty.");
+    }
```

Comment:
I think it would be better to change these methods into classes and package them under `DukeException`. Otherwise, you can simply `throw new DukeException()` with the corresponding error message.